src/lib/renderComponent.tsx

```diff
@@ -48,7 +71,9 @@ const renderComponent = <P extends {}>(
         const classes = getClasses(renderer, props, rules, mergedVariables, theme)
         classes.root = cx(className, classes.root, props.className)
 
-        const config: IRenderResultConfig<P> = { ElementType, rest, classes }
+        const accessibility = getAccessibility(props, state, defaultAccessibility)
```

Comment:
given that this function makes decision about accessibility strategy basing on `props` and `state` - essentially, the same data that is available at the rendering phase, - we could eliminate this call here and introduce all the necessary selection logic to accessibility behavior itself